providers/jackson2/src/main/java/org/jboss/resteasy/plugins/providers/jackson/PatchMethodFilter.java

```diff
@@ -32,36 +32,39 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.github.fge.jsonpatch.JsonPatch;
 import com.github.fge.jsonpatch.JsonPatchException;
+import com.github.fge.jsonpatch.mergepatch.JsonMergePatch;
+
 /*
 * @author <a href="mailto:ema@redhat.com">Jim Ma</a>
 */
 @Provider
 @Priority(Integer.MAX_VALUE)
 public class PatchMethodFilter implements ContainerRequestFilter
 {
+   //TODO:thse should go to jaxrs spec apis
+   public static final String APPLICATION_JSON_MERGE_PATCH_JSON = "application/merge-patch+json";
+
+   public static final MediaType APPLICATION_JSON_MERGE_PATCH_JSON_TYPE = new MediaType("application",
+         "merge-patch+json");
+
    @Override
-   @SuppressWarnings({"rawtypes", "unchecked"})
+   @SuppressWarnings(
+   {"rawtypes", "unchecked"})
    public void filter(ContainerRequestContext requestContext) throws IOException
    {
-      //Strict the filter is only executed for patch method and media type is APPLICATION_JSON_PATCH_JSON_TYPE
       if (requestContext.getMethod().equals("PATCH")
-            && MediaType.APPLICATION_JSON_PATCH_JSON_TYPE.isCompatible(requestContext.getMediaType()))
+            && MediaType.APPLICATION_JSON_PATCH_JSON_TYPE.isCompatible(requestContext.getMediaType())
+            || requestContext.getMethod().equals("PATCH")
+            && APPLICATION_JSON_MERGE_PATCH_JSON_TYPE.isCompatible(requestContext.getMediaType()))
```

Comment:
Can you rework the if conditions so that requestContext.getMethod().equals("PATCH") is executed once?